### DISCUSSION TITLE
Default models and data?

### DISCUSSION THREAD

**Discussion by ewancarr** [pmsims-package/pmsims#25] [Ideas]
Hi all,

Now that we have a better idea of the general workflow, it'd be good to agree the different model/data types we're going to offer 'out-of-the-box'. For example:

## Models

1. LASSO linear regression
2. LASSO linear regression
3. LASSO Cox regression

## Data

1. 'Genetic' profile (lots of binary predictors)
2. 'Clinical' profile (mix of binary and continuous, <100 parameters)
3. 'High-dimensional' profile (mix, >500 parameters)

When generating data, we'll need to simulate the correct type of y — e.g., linear, binary, or Cox. This could be done automatically, based on the chosen model.

***

We then would need to decide:

* For models:
  i. Evaluation metrics
  ii. Defaults for tuning parameters.
* For data:
  i. Composition (i.e., default number/type of predictors)
  ii. tuning parameters.

(I'm probably forgetting some options).

The idea is that users can use these defaults without specifying their model/data functions, e.g.

```
calculate_sample_size(model = "binary",
                      data = list(type = "genetic",
                                  n_predictors = 200,
                                  signal = 0.5),
                      metric = "auc",
                      expectation = 0.7,
                      ...)
                                   
```

What do you think?

**Comment by GForb** [comment]
Playing around with this I think our default should be continuous predictors, N(0,1) - Binary predictors require a predictor prevalence to be specified which is another thing to think about.

**Comment by GForb** [comment]
I also think we should add ridge as well as Lasso - as felix's simulations showed Lasso can lead to larger sample size when all predictors are true predictors - as our default data generating mechanisms have all true predictors we should offer a penalisation option out of the box that reduces the sample size.

**Comment by GForb** [comment]
With the default data generating models the number of predictors should be a easily accessible argument - this is probably the only thing that analysts will really have control over.
